### PR TITLE
Refactor: implement the HandlerWrapper and decouple server/agent interface from implementation 

### PR DESCRIPTION
**Pull request by charleszheng44**
1. implement the handlerwrapper, which takes in one http.handler and wraps it within another
2. decouple the anp tunnel server/agent using TunnelServer and TunnelAgent interface

### CODE REVIEW

File: pkg/yurttunnel/server/handlerwrapper.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	})
+}
+
+// WrapWrapHandler wraps the coreHandler with chains
+func WrapHandler(coreHandler http.Handler, middlewares ...Middleware) http.Handler {
+	handler := coreHandler
+	if len(middlewares) == 0 {
+		return handler
+	}
+	for i := len(middlewares) - 1; i > 0; i-- {
```
Comment (rambohe-ch) [inline pkg/yurttunnel/server/handlerwrapper.go:48]: if condition miss i=0, so middlewares[0] can not reach

File: pkg/yurttunnel/agent/agent.go [hunk 2]
```diff
@@ -35,6 +35,34 @@ import (
 	"github.com/alibaba/openyurt/pkg/yurttunnel/pki/certmanager"
 )
 
+// TunnelAgent sets up tunnel to TunnelServer, receive requests
+// from tunnel, and forwards requests to kubelet
+type TunnelAgent interface {
```
Comment (rambohe-ch) [inline pkg/yurttunnel/agent/agent.go:40]: now we make up interface for tunnel agent and tunnel server, it is better to separate the interface implementation(for example: anp tunnel) into independent dir or file.

File: pkg/yurttunnel/server/handlerwrapper.go (hunk 1 above)
Comment (charleszheng44) [reply pkg/yurttunnel/server/handlerwrapper.go]: fixed

File: pkg/yurttunnel/agent/agent.go (hunk 2 above)
Comment (charleszheng44) [reply pkg/yurttunnel/agent/agent.go:35]: create independent files for them

**Comment by rambohe-ch** [comment]
lgtm

**Pull request merged**